pkg/chunk/cache/memcached.go

```diff
@@ -146,19 +146,20 @@ func (c *Memcached) Fetch(ctx context.Context, keys []string) (found []string, b
 
 func (c *Memcached) fetch(ctx context.Context, keys []string) (found []string, bufs [][]byte, missed []string) {
 	var items map[string]*memcache.Item
-	err := instr.CollectedRequest(ctx, "Memcache.GetMulti", c.requestDuration, memcacheStatusCode, func(innerCtx context.Context) error {
-		sp := opentracing.SpanFromContext(innerCtx)
-		sp.LogFields(otlog.Int("keys requested", len(keys)))
+	const method = "Memcache.GetMulti"
+	err := instr.CollectedRequest(ctx, method, c.requestDuration, memcacheStatusCode, func(innerCtx context.Context) error {
+		log, _ := spanlogger.New(innerCtx, method)
+		log.LogFields(otlog.Int("keys requested", len(keys)))
 
 		var err error
 		items, err = c.memcache.GetMulti(keys)
 
-		sp.LogFields(otlog.Int("keys found", len(items)))
+		log.LogFields(otlog.Int("keys found", len(items)))
 
 		// Memcached returns partial results even on error.
 		if err != nil {
-			sp.LogFields(otlog.Error(err))
-			level.Error(c.logger).Log("msg", "Failed to get keys from memcached", "err", err)
+			log.Error(err) //nolint:errcheck
```

Comment:
I couldn't get that to work properly, for various variations of `(import path.Struct).Function`. If you have a suggestion for what to use as the entry in that file let me know.

Reply:
Please see the latest commit as well, if we want to have the `ctx` assignment in the bigtable index client for example, we need nolint comments again.

Reply:
I think `(*github.com/cortexproject/cortex/pkg/util/spanlogger.SpanLogger).Error` should work (it does with standalone errcheck utility). When I update **`golangci-lint` in the build image to 1.30.0**, I no longer see this warning, even without changing .errcheck-exclude. 😕 

We could also remove error return value from this method, since no production code uses it.